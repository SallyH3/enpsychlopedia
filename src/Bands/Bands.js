import React, { Component } from 'react';
import Band from '../Band/Band.js';

export default class Bands extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: [],
      inactive: []
    }
  }

  assignActiveBands() {
    let activeBands = this.props.genreBands.filter(band => {
        return band.active === true;
      })
    this.setState({active: activeBands})
  }

  assignInactiveBands() {
    let inactiveBands = this.props.genreBands.filter(band => {
        return band.active === false;
      })
    this.setState({inactive: inactiveBands})
  }

  componentWillReceiveProps() {
    this.assignActiveBands();
    this.assignInactiveBands();
  }

  filterAllBands(props) {
    console.log(props)
    return (
      {this.props.genreBands.map((band, index) => (
        <Band key={index}
               bandName={band.name}
               bandImg={band.image}
        />
      ))}
    )
  }

  filterActiveBands() {
    return(
      {this.state.active.map((band, index) => (
        <Band key={index}
                bandName={band.name}
                bandImg={band.image}
            />
          ))}
    )
  }

  filterInactiveBands() {
    return(
      {this.state.inactive.map((band, index) => (
        <Band key={index}
               bandName={band.name}
               bandImg={band.image}
        />
      ))}
    )
  }

  render() { 
  let fallback = this.props.currentSelection;
  if (!fallback) { 
    return (
      <h1>Hi Sally!</h1>
    )
  }
    return (
      <section className='bands-wrapper'>
        if(this.props.currentSelection === 'all') {
          {this.filterAllBands(this.props)}
        } else if (this.props.currentSelection === 'active') {
          {this.filterActiveBands()}
        } else {
          {this.filterInactiveBands()}
        }
      </section>
    )
  }
}