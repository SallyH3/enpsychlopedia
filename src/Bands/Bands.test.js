import React from 'react';
import Bands from './Bands';
import { shallow } from 'enzyme';
let mimicBands = [{ name: 'Sparklehorse',
origin: 'Richmond Virginia',
image: 'https://i.imgur.com/ciMF4gb.jpg',
members: [ 'Mark Linkous', 'Scott Minor' ],
genreId: 1,
active: false,
label: 'Slow River',
sideProjects: 
 [ 'Danger Mouse',
   'Sol Seppy',
   'David Lynch',
   'Angela Faye Martin' ] },
{ name: 'Quest for Fire',
origin: 'Toronto Canada',
image: 'https://i.imgur.com/lamtwzr.jpg',
members: 
 [ 'Andrew Moszynski',
   'Chad Ross',
   'Mike Maxymuik',
   'Josh Bauman' ],
genreId: 1,
active: false,
label: 'Tee Pee',
sideProjects: [ 'The Deadly Snakes', 'Cursed', 'Comet Control' ] },
{ name: 'The Dream Syndicate',
origin: 'Los Angeles California',
image: 'https://i.imgur.com/7zklXlb.jpg',
members: [ 'Steve Wynn', 'Dennis Duck', 'Mark Walton', 'Jason Victor' ],
genreId: 1,
active: true,
label: 'Big Time',
sideProjects: 
 [ 'Grave',
   'Continental Drifters',
   'Steve Wynn and the Miracle Three',
   'Gutterball' ] },
{ name: 'Ozric Tentacles',
origin: 'Somerset England',
image: 'https://i.imgur.com/ClePm63.jpg',
members: [ 'Ed Wynne', 'Brandi Wynne', 'Silas Neptune', 'Balazs Szende' ],
genreId: 1,
active: true,
label: 'Magna Carta',
sideProjects: 
 [ 'Eat Static',
   'Nodens Ictus',
   'Here and Now',
   'Invisible System' ] },
{ name: 'Stardeath and the White Dwarfs',
origin: 'Oklahoma City Oklahoma',
image: 'https://i.imgur.com/6zAdp1g.jpg',
members: 
 [ 'Dennis Coyne',
   'Casey Joseph',
   'Matt Duckworth',
   'Ford Chastain' ],
genreId: 1,
active: true,
label: 'Warner Bros',
sideProjects: 
 [ 'The Flaming Lips',
   'New Fumes',
   'Linear Downfall',
   'Spaceface' ] },
{ name: 'The Black Angels',
origin: 'Austin TX',
image: 'https://i.imgur.com/TdwPPTe.jpg',
members: 
 [ 'Stephanie Bailey',
   'Christian Bland',
   'Alex Maas',
   'Kyle Hunt',
   'Jake Garcia' ],
genreId: 1,
active: true,
label: 'Light in the Attic',
sideProjects: 
 [ 'Mien',
   'The Viet Minh',
   'Christian Bland and The Revelators',
   'The UFO Club',
   'Sweet Tea' ] },
{ name: 'Spacemen 3',
origin: 'Warwickshire England',
image: 'https://i.imgur.com/2UcOZah.jpg',
members: 
 [ 'Peter Kember',
   'Jason Pierce',
   'Tim Morris',
   'Pete Bain',
   'Natty Brooker',
   'Sterling Roswell',
   'Will Carruthers',
   'Jonny Mattock',
   'Mark Refoy' ],
genreId: 1,
active: false,
label: 'Dedicated Records',
sideProjects: [ 'Spectrum', 'Spiritualized', 'Darkside' ] },
{ name: 'The Teardrops',
origin: 'Liverpool England',
image: 'https://i.imgur.com/hk7Fasw.png',
members: 
 [ 'Julian Cope',
   'Mick Finkler',
   'Gary Dwyer',
   'Paul Simpson',
   'Ged Quinn',
   'David Balfe',
   'Alan Gill',
   'Alfie Agius',
   'Jeff Hammer',
   'Troy Tate',
   'Ronnie Francois',
   'Ted Emmett',
   'Andy Radek' ],
genreId: 1,
active: false,
label: 'Zoo Records',
sideProjects: 
 [ 'Julian Cope',
   'Crucial Three',
   'Big In Japan',
   'Dalek I Love You',
   'The Wild Swans' ] },
{ name: 'Spindrift',
origin: 'Los Angeles CA',
image: 'https://i.imgur.com/JdLZXQX.jpg',
members: 
 [ 'Kirpatrick Thomas',
   'James Acton',
   'Daniel Allaire',
   'Joe Baluta',
   'Bobby Bones',
   'Becca Davidson',
   'Julie Patterson' ],
genreId: 1,
active: true,
label: 'Tee Pee Records',
sideProjects: [ 'Brian Jonestown Massacre', 'The Warlocks', 'Crooked Cowboy' ] },
{ name: 'Echo and the Bunnymen',
origin: 'Liverpool England',
image: 'https://i.imgur.com/suHhm1z.jpg',
members: [ 'Ian McCulloch', 'Will Sergeant' ],
genreId: 1,
active: true,
label: 'Warner Bros',
sideProjects: [ 'Electrafixion', 'Youth' ] },
{ name: 'The Church',
origin: 'Sydney Australia',
image: 'https://i.imgur.com/XoF7K90.jpg',
members: [ 'Steve Kilbey', 'Peter Koppes', 'Tim Powles', 'Ian Haug' ],
genreId: 1,
active: true,
label: 'Capitol',
sideProjects: [ 'Tactics', 'The Crystal Set', 'The Reels', 'Hex', 'The Well' ]}];

describe('Bands', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Bands  
      genreBands={mimicBands}
      currentSelection={'all'}
      />
    )
  });
  
  it('should match snapshot', () => {
    wrapper.setProps({currentSelection: 'all', genreBands: mimicBands});
    expect(wrapper).toMatchSnapshot("first");
  });

  it('should match snapshot with active set as filter', () => {
    wrapper.setProps({currentSelection: 'active', genreBands: mimicBands});
    expect(wrapper).toMatchSnapshot("second");
  });

  it('should match snapshot with inactive set as filter', () => {
    wrapper.setProps({currentSelection: 'inactive', genreBands: mimicBands});
    expect(wrapper).toMatchSnapshot("third");
  });

  it('should have default states', () => {
    expect(wrapper.state()).toEqual({ 
      active: [],
      inactive: [],
      currentPopup: {},
      showPopup: false
    });
  });
});