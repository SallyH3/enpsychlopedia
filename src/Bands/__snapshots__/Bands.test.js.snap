// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bands should match snapshot with active set as filter: second 1`] = `
<section
  className="bands-wrapper"
>
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/7zklXlb.jpg",
        "label": "Big Time",
        "members": Array [
          "Steve Wynn",
          "Dennis Duck",
          "Mark Walton",
          "Jason Victor",
        ],
        "name": "The Dream Syndicate",
        "origin": "Los Angeles California",
        "sideProjects": Array [
          "Grave",
          "Continental Drifters",
          "Steve Wynn and the Miracle Three",
          "Gutterball",
        ],
      }
    }
    fetchBand={[Function]}
    key="0"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/ClePm63.jpg",
        "label": "Magna Carta",
        "members": Array [
          "Ed Wynne",
          "Brandi Wynne",
          "Silas Neptune",
          "Balazs Szende",
        ],
        "name": "Ozric Tentacles",
        "origin": "Somerset England",
        "sideProjects": Array [
          "Eat Static",
          "Nodens Ictus",
          "Here and Now",
          "Invisible System",
        ],
      }
    }
    fetchBand={[Function]}
    key="1"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/6zAdp1g.jpg",
        "label": "Warner Bros",
        "members": Array [
          "Dennis Coyne",
          "Casey Joseph",
          "Matt Duckworth",
          "Ford Chastain",
        ],
        "name": "Stardeath and the White Dwarfs",
        "origin": "Oklahoma City Oklahoma",
        "sideProjects": Array [
          "The Flaming Lips",
          "New Fumes",
          "Linear Downfall",
          "Spaceface",
        ],
      }
    }
    fetchBand={[Function]}
    key="2"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/TdwPPTe.jpg",
        "label": "Light in the Attic",
        "members": Array [
          "Stephanie Bailey",
          "Christian Bland",
          "Alex Maas",
          "Kyle Hunt",
          "Jake Garcia",
        ],
        "name": "The Black Angels",
        "origin": "Austin TX",
        "sideProjects": Array [
          "Mien",
          "The Viet Minh",
          "Christian Bland and The Revelators",
          "The UFO Club",
          "Sweet Tea",
        ],
      }
    }
    fetchBand={[Function]}
    key="3"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/JdLZXQX.jpg",
        "label": "Tee Pee Records",
        "members": Array [
          "Kirpatrick Thomas",
          "James Acton",
          "Daniel Allaire",
          "Joe Baluta",
          "Bobby Bones",
          "Becca Davidson",
          "Julie Patterson",
        ],
        "name": "Spindrift",
        "origin": "Los Angeles CA",
        "sideProjects": Array [
          "Brian Jonestown Massacre",
          "The Warlocks",
          "Crooked Cowboy",
        ],
      }
    }
    fetchBand={[Function]}
    key="4"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/suHhm1z.jpg",
        "label": "Warner Bros",
        "members": Array [
          "Ian McCulloch",
          "Will Sergeant",
        ],
        "name": "Echo and the Bunnymen",
        "origin": "Liverpool England",
        "sideProjects": Array [
          "Electrafixion",
          "Youth",
        ],
      }
    }
    fetchBand={[Function]}
    key="5"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/XoF7K90.jpg",
        "label": "Capitol",
        "members": Array [
          "Steve Kilbey",
          "Peter Koppes",
          "Tim Powles",
          "Ian Haug",
        ],
        "name": "The Church",
        "origin": "Sydney Australia",
        "sideProjects": Array [
          "Tactics",
          "The Crystal Set",
          "The Reels",
          "Hex",
          "The Well",
        ],
      }
    }
    fetchBand={[Function]}
    key="6"
  />
</section>
`;

exports[`Bands should match snapshot with inactive set as filter: third 1`] = `
<section
  className="bands-wrapper"
>
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/ciMF4gb.jpg",
        "label": "Slow River",
        "members": Array [
          "Mark Linkous",
          "Scott Minor",
        ],
        "name": "Sparklehorse",
        "origin": "Richmond Virginia",
        "sideProjects": Array [
          "Danger Mouse",
          "Sol Seppy",
          "David Lynch",
          "Angela Faye Martin",
        ],
      }
    }
    fetchBand={[Function]}
    key="0"
  />
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/lamtwzr.jpg",
        "label": "Tee Pee",
        "members": Array [
          "Andrew Moszynski",
          "Chad Ross",
          "Mike Maxymuik",
          "Josh Bauman",
        ],
        "name": "Quest for Fire",
        "origin": "Toronto Canada",
        "sideProjects": Array [
          "The Deadly Snakes",
          "Cursed",
          "Comet Control",
        ],
      }
    }
    fetchBand={[Function]}
    key="1"
  />
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/2UcOZah.jpg",
        "label": "Dedicated Records",
        "members": Array [
          "Peter Kember",
          "Jason Pierce",
          "Tim Morris",
          "Pete Bain",
          "Natty Brooker",
          "Sterling Roswell",
          "Will Carruthers",
          "Jonny Mattock",
          "Mark Refoy",
        ],
        "name": "Spacemen 3",
        "origin": "Warwickshire England",
        "sideProjects": Array [
          "Spectrum",
          "Spiritualized",
          "Darkside",
        ],
      }
    }
    fetchBand={[Function]}
    key="2"
  />
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/hk7Fasw.png",
        "label": "Zoo Records",
        "members": Array [
          "Julian Cope",
          "Mick Finkler",
          "Gary Dwyer",
          "Paul Simpson",
          "Ged Quinn",
          "David Balfe",
          "Alan Gill",
          "Alfie Agius",
          "Jeff Hammer",
          "Troy Tate",
          "Ronnie Francois",
          "Ted Emmett",
          "Andy Radek",
        ],
        "name": "The Teardrops",
        "origin": "Liverpool England",
        "sideProjects": Array [
          "Julian Cope",
          "Crucial Three",
          "Big In Japan",
          "Dalek I Love You",
          "The Wild Swans",
        ],
      }
    }
    fetchBand={[Function]}
    key="3"
  />
</section>
`;

exports[`Bands should match snapshot: first 1`] = `
<section
  className="bands-wrapper"
>
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/ciMF4gb.jpg",
        "label": "Slow River",
        "members": Array [
          "Mark Linkous",
          "Scott Minor",
        ],
        "name": "Sparklehorse",
        "origin": "Richmond Virginia",
        "sideProjects": Array [
          "Danger Mouse",
          "Sol Seppy",
          "David Lynch",
          "Angela Faye Martin",
        ],
      }
    }
    fetchBand={[Function]}
    key="0"
  />
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/lamtwzr.jpg",
        "label": "Tee Pee",
        "members": Array [
          "Andrew Moszynski",
          "Chad Ross",
          "Mike Maxymuik",
          "Josh Bauman",
        ],
        "name": "Quest for Fire",
        "origin": "Toronto Canada",
        "sideProjects": Array [
          "The Deadly Snakes",
          "Cursed",
          "Comet Control",
        ],
      }
    }
    fetchBand={[Function]}
    key="1"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/7zklXlb.jpg",
        "label": "Big Time",
        "members": Array [
          "Steve Wynn",
          "Dennis Duck",
          "Mark Walton",
          "Jason Victor",
        ],
        "name": "The Dream Syndicate",
        "origin": "Los Angeles California",
        "sideProjects": Array [
          "Grave",
          "Continental Drifters",
          "Steve Wynn and the Miracle Three",
          "Gutterball",
        ],
      }
    }
    fetchBand={[Function]}
    key="2"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/ClePm63.jpg",
        "label": "Magna Carta",
        "members": Array [
          "Ed Wynne",
          "Brandi Wynne",
          "Silas Neptune",
          "Balazs Szende",
        ],
        "name": "Ozric Tentacles",
        "origin": "Somerset England",
        "sideProjects": Array [
          "Eat Static",
          "Nodens Ictus",
          "Here and Now",
          "Invisible System",
        ],
      }
    }
    fetchBand={[Function]}
    key="3"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/6zAdp1g.jpg",
        "label": "Warner Bros",
        "members": Array [
          "Dennis Coyne",
          "Casey Joseph",
          "Matt Duckworth",
          "Ford Chastain",
        ],
        "name": "Stardeath and the White Dwarfs",
        "origin": "Oklahoma City Oklahoma",
        "sideProjects": Array [
          "The Flaming Lips",
          "New Fumes",
          "Linear Downfall",
          "Spaceface",
        ],
      }
    }
    fetchBand={[Function]}
    key="4"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/TdwPPTe.jpg",
        "label": "Light in the Attic",
        "members": Array [
          "Stephanie Bailey",
          "Christian Bland",
          "Alex Maas",
          "Kyle Hunt",
          "Jake Garcia",
        ],
        "name": "The Black Angels",
        "origin": "Austin TX",
        "sideProjects": Array [
          "Mien",
          "The Viet Minh",
          "Christian Bland and The Revelators",
          "The UFO Club",
          "Sweet Tea",
        ],
      }
    }
    fetchBand={[Function]}
    key="5"
  />
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/2UcOZah.jpg",
        "label": "Dedicated Records",
        "members": Array [
          "Peter Kember",
          "Jason Pierce",
          "Tim Morris",
          "Pete Bain",
          "Natty Brooker",
          "Sterling Roswell",
          "Will Carruthers",
          "Jonny Mattock",
          "Mark Refoy",
        ],
        "name": "Spacemen 3",
        "origin": "Warwickshire England",
        "sideProjects": Array [
          "Spectrum",
          "Spiritualized",
          "Darkside",
        ],
      }
    }
    fetchBand={[Function]}
    key="6"
  />
  <Band
    band={
      Object {
        "active": false,
        "genreId": 1,
        "image": "https://i.imgur.com/hk7Fasw.png",
        "label": "Zoo Records",
        "members": Array [
          "Julian Cope",
          "Mick Finkler",
          "Gary Dwyer",
          "Paul Simpson",
          "Ged Quinn",
          "David Balfe",
          "Alan Gill",
          "Alfie Agius",
          "Jeff Hammer",
          "Troy Tate",
          "Ronnie Francois",
          "Ted Emmett",
          "Andy Radek",
        ],
        "name": "The Teardrops",
        "origin": "Liverpool England",
        "sideProjects": Array [
          "Julian Cope",
          "Crucial Three",
          "Big In Japan",
          "Dalek I Love You",
          "The Wild Swans",
        ],
      }
    }
    fetchBand={[Function]}
    key="7"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/JdLZXQX.jpg",
        "label": "Tee Pee Records",
        "members": Array [
          "Kirpatrick Thomas",
          "James Acton",
          "Daniel Allaire",
          "Joe Baluta",
          "Bobby Bones",
          "Becca Davidson",
          "Julie Patterson",
        ],
        "name": "Spindrift",
        "origin": "Los Angeles CA",
        "sideProjects": Array [
          "Brian Jonestown Massacre",
          "The Warlocks",
          "Crooked Cowboy",
        ],
      }
    }
    fetchBand={[Function]}
    key="8"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/suHhm1z.jpg",
        "label": "Warner Bros",
        "members": Array [
          "Ian McCulloch",
          "Will Sergeant",
        ],
        "name": "Echo and the Bunnymen",
        "origin": "Liverpool England",
        "sideProjects": Array [
          "Electrafixion",
          "Youth",
        ],
      }
    }
    fetchBand={[Function]}
    key="9"
  />
  <Band
    band={
      Object {
        "active": true,
        "genreId": 1,
        "image": "https://i.imgur.com/XoF7K90.jpg",
        "label": "Capitol",
        "members": Array [
          "Steve Kilbey",
          "Peter Koppes",
          "Tim Powles",
          "Ian Haug",
        ],
        "name": "The Church",
        "origin": "Sydney Australia",
        "sideProjects": Array [
          "Tactics",
          "The Crystal Set",
          "The Reels",
          "Hex",
          "The Well",
        ],
      }
    }
    fetchBand={[Function]}
    key="10"
  />
</section>
`;
